packages/aws-cdk/test/integ/run-against-dist.bash

```diff
@@ -22,17 +22,53 @@ function serve_npm_packages() {
     return
   fi
 
-  tarballs=$dist_root/js/*.tgz
+  if [ ! -z ${USE_PUBLISHED_FRAMEWORK_VERSION:-} ]; then
 
-  log "Discovering local package names..."
-  # Read the package names from each tarball, so that we can generate
-  # a Verdaccio config that will keep each of these packages locally
-  # and not go to NPMJS for it.
-  package_names=""
-  for tgz in $tarballs; do
-    name=$(node -pe 'JSON.parse(process.argv[1]).name' "$(tar xOzf $tgz package/package.json)")
-    package_names="$package_names $name"
-  done
+    echo "Testing against latest published versions of the framework"
+
+    # when using latest published framework, only
+    # install the cli and its dependencies.
+
+    cli_root=./package  
+
+    version=$(node -p "require('${cli_root}/package.json').version")
+
+    # good lord
+    echo "Fetching CLI dependencies"
+    cli_deps=$(node -p "Object.entries(require('${cli_root}/package.json').dependencies).filter(x => x[0].includes('@aws-cdk')).map(x => x[0].replace('@aws-cdk/', '')).join(' ')")
+
+    tarballs=$dist_root/js/aws-cdk-${version}.tgz
+    package_names="aws-cdk"
+
+    for dep in ${cli_deps}; do
+      tarball=$dist_root/js/${dep}@${version}.jsii.tgz
+      package=@aws-cdk/${dep}
+      if [ ! -f ${tarball} ]; then
+        # not a jsii dependency, the tarball is different...
+        tarball=$dist_root/js/aws-cdk-${dep}-${version}.tgz        
+      fi 
+
+      tarballs="${tarballs} ${tarball}"
+      package_names="${package_names} ${package}"
+    done
+
+  else
+
+    echo "Testing against local versions of the framework"
+
+    tarballs=$dist_root/js/*.tgz
+
+    log "Discovering local package names..."
+    # Read the package names from each tarball, so that we can generate
+    # a Verdaccio config that will keep each of these packages locally
```

Comment:
I am curious why we need to go through verdaccio? Can't we just `npm install $tarballs`? It it contains all the deps of the CLI, it will work.

Reply:
I guess thats true. But i'm not sure if there is big benefit in that if we still need the ugly parsing of the package.json file to extract the CLI dependencies.

```bash
# good lord
echo "Fetching CLI dependencies"
cli_deps=$(node -p "Object.entries(require('${cli_root}/package.json').dependencies).filter(x => x[0].includes('@aws-cdk')).map(x => x[0].replace('@aws-cdk/', '')).join(' ')")
```

We could theoretically skip the parsing and simply `npm install` all tarballs and thats it. Assuming that subsequent `npm install` calls will override any installed packages (will it? it will be the same version..)

Also, running `npm install` at this stage bares the risk of not working if the tests are executed from a different directory (tree).

How strongly do you feel about this? I'd like not rattle the boat here too much.